### PR TITLE
uct/dc_mlx5: added procesing of GRH

### PR DESCRIPTION
**Pull request by hoopoepg**
- added processing of GRH to DC MLX5 transport
- removed limitations on local connection only

### CODE REVIEW

**Comment by hoopoepg** [comment]
bot:mlx:retest

File: src/uct/ib/rc/accel/rc_mlx5_common.h [hunk 1]
```diff
@@ -275,7 +275,8 @@ uct_rc_mlx5_txqp_inline_post(uct_rc_iface_t *iface, enum ibv_qp_type qp_type,
                              unsigned opcode, const void *buffer, unsigned length,
                   /* SEND */ uint8_t am_id, uint64_t am_hdr, uint32_t imm_val_be,
                   /* RDMA */ uint64_t rdma_raddr, uct_rkey_t rdma_rkey,
-                  /* AV   */ uct_ib_mlx5_base_av_t *av, size_t av_size
+                  /* AV   */ uct_ib_mlx5_base_av_t *av, size_t av_size,
```
Comment (yosefe) [inline src/uct/ib/rc/accel/rc_mlx5_common.h]: i would rename/order this as follows for better consistency:
```
uct_ib_mlx5_base_av_t *base_av,  
struct mlx5_grh_av *grh_av,
size_t av_size
```

File: src/uct/ib/dc/accel/dc_mlx5.c [hunk 2]
```diff
@@ -606,7 +590,8 @@ ucs_status_t uct_dc_mlx5_ep_fc_ctrl(uct_ep_t *tl_ep, unsigned op,
                                      txqp, txwq, MLX5_OPCODE_SEND,
                                      &av /*dummy*/, 0, op, sender_ep, 0,
                                      0, 0,
-                                     &av, uct_ib_mlx5_wqe_av_size(&av));
+                                     &av, uct_ib_mlx5_wqe_av_size(&av),
+                                     NULL); /* TODO: add GRH processing here */
```
Comment (yosefe) [inline src/uct/ib/dc/accel/dc_mlx5.c]: need to complete this to make FC work with GRH..

Comment (hoopoepg) [reply src/uct/ib/dc/accel/dc_mlx5.c]: it will be another PR (it will affect dc/verbs too)

Comment (yosefe) [reply src/uct/ib/dc/accel/dc_mlx5.c]: ok

**Comment by brminich** [comment]
:+1:

File: src/uct/ib/rc/accel/rc_mlx5_common.h (hunk 1 above)
Comment (hoopoepg) [reply src/uct/ib/rc/accel/rc_mlx5_common.h]: done

**Comment by yosefe** [comment]
bot:bgate:retest

**Comment by hoopoepg** [comment]
bot:retest

**Comment by hoopoepg** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4546/ for details (Mellanox internal link).

**Comment by hoopoepg** [comment]
bot:mlx:retest

**Comment by hoopoepg** [comment]
bot:bgate:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4549/ for details (Mellanox internal link).

**Comment by hoopoepg** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4551/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4552/ for details (Mellanox internal link).

**Pull request merged**